src/main/java/com/example/information/KingInformationEngine.java

```diff
@@ -0,0 +1,53 @@
+package com.example.information;
+
+import com.brunomnsilva.smartgraph.graph.Vertex;
+import com.example.algorithm.AlgorithmPhase;
+import com.example.algorithm.AlgorithmType;
+import com.example.algorithm.VertexRole;
+import com.example.algorithm.report.StepReport;
+import com.example.information.printer.InformationPrinter;
+import lombok.AllArgsConstructor;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@AllArgsConstructor
+public class KingInformationEngine implements InformationEngine {
+    private InformationPrinter informationPrinter;
+
+    @Override
+    public void processReport(StepReport stepReport) {
+        informationPrinter.setAlgorithmName(AlgorithmType.KING.name());
+        informationPrinter.setAlgorithmPhase(stepReport.getAlgorithmPhase());
+        informationPrinter.setStepDescription(generateDescription(stepReport));
+        informationPrinter.listProperties(generateProperties(stepReport));
+        informationPrinter.renderView();
+    }
+
+    private Map<String, String> generateProperties(StepReport stepReport) {
+        Map<String, String> properties = stepReport.getProperties();
+        Map<String, String> propertiesToSend = new HashMap<>();
+        propertiesToSend.put("phase", properties.get("phase"));
+        Optional<Map.Entry<Vertex<Integer>, VertexRole>> kingEntry = stepReport.getRoles().entrySet()
+                .stream()
+                .filter(entry -> entry.getValue() == VertexRole.KING)
+                .findFirst();
+        kingEntry.ifPresent(entry -> {
```

Comment:
How about we add `king: -` instead of not including that property at all so the view is more static?

Reply:
Fixed